### PR TITLE
doc: Use `Signal.emit` instead of `emit_signal` in Object examples

### PR DESCRIPTION
**Pull request by kkoang**
Change the code and use the new method of godot4

URL:https://docs.godotengine.org/en/latest/classes/class_object.html

### CODE REVIEW

**Comment by YuriSizov** [comment]
> This PR tries changes the function shown in the examples of `Object.emit_signal()` itself to use `Signal.emit(`) which is not what is being described.

Well, we're using Signal's API for connecting and disconnecting, for example. So I think this change is valid.

> this entire set of paragraphs should be stripped away instead, and moved somewhere else, at best. The entire thing is needlessly long and verbose.

You are probably right there.

**Review by mhilbrunner — APPROVED** [review]
I think this change is good by itself, regardless of possible further improvements of the whole section. Moving away from strings in favor of signal names wherever we can in examples is good, IMO.

Unfortunately, this needs a rebase.

**Comment by akien-mga** [comment]
Rebased, and gave the PR and commit a meaningful title.

Would be good that @godotengine/dotnet checks that the C# example is correct.

**Comment by paulloz** [comment]
Hi. Nope, this does not work in C#. Currently, the only way to emit signals is through `EmitSignal` (see https://docs.godotengine.org/en/latest/tutorials/scripting/c_sharp/c_sharp_signals.html#signal-emission). There are WIPs to have access to a strongly typed API (#68233), but it is not merged at the moment.

**Comment by kkoang** [comment]
@paulloz I restored the C# code to its original state. Sorry, I’m not familiar with C#.

File: doc/classes/Object.xml [hunk 1]
```diff
@@ -556,7 +556,7 @@
...
 				[b]Binding and passing parameters:[/b]
 				The syntax to bind parameters is through [method Callable.bind], which returns a copy of the [Callable] with its parameters bound.
-				When calling [method emit_signal], the signal parameters can be also passed. The examples below show the relationship between these signal parameters and bound parameters.
+				When calling [method Signal.emit], the signal parameters can be also passed. The examples below show the relationship between these signal parameters and bound parameters.
```
[the remark this answers is not in the document]
Comment (kkoang) [reply doc/classes/Object.xml:559]: OK, thanks

**Comment by akien-mga** [comment]
Looks good! Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by kkoang** [comment]
@akien-mga I am not a professional programmer. I have been reading [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) and learning git, and it may take a few days. If you have already helped me complete the "rebase", I am very grateful.

**Comment by akien-mga** [comment]
Yes it's all good now :) I'll merge this soon.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

And apologies for the delay getting this reviewed.

**Pull request merged**